src/KernelLaunch.jl

```diff
@@ -135,6 +155,8 @@ import KernelAbstractions.NDIteration.StaticSize
         fun(args..., offset)
         bc!(arch, grid, bc)
     else
+        foreach(arg -> apply_disable_task_sync!(backend, arg, disable_task_sync!), args)
```

Comment:
What if we check on the fly for the args passed to the `launch_with_bc` function and disable task sync recursively there on any `Field` types, and enable task sync on any `Field` types before returning from the kernel launch function?

Reply:
This would still allow for using different task per BC kernel and avoid a heavy rework of workers launching workers and possibly paying the price of serialising all the compute first.